semantic_inference_ros/app/instance_segmentation_node

```diff
@@ -0,0 +1,236 @@
+            self.config = sc.Config.load(InstanceSegmentationNodeConfig, config_path)
+
+        self.get_logger().info(f"Initializing with {self.config.show()}")
+        device = models.default_device()
+        self._model = models.InstanceSegmenter(self.config.model).to(device)
+        self._model.eval()  # TODO: causing issue with yolo model
+        self.get_logger().info("Finished initializing!")
+
+        self._pub = self.create_publisher(
+            Image, "semantic/image_raw", 1
+        )  # publish segmented image
+        self._worker = semantic_inference_ros.ImageWorker(
+            self, self.config.worker, "color/image_raw", self._spin_once
+        )  # put image in queue for processing
+
+        self._visualizer = "place_holder"
+        # self._visualizer = self.config.visualizer.create()
+        if self._visualizer is not None:
+            # TODO: write proper visualizer for instance segmentation
+            self._overlay_pub = self.create_publisher(
+                Image, "semantic_overlay/image_raw", 1
+            )
+
+    def _spin_once(self, header, img):
+        with torch.no_grad():
+            ret = self._model.segment(img, is_rgb_order=True).cpu()
+
+        if ret.masks is None:
+            self.get_logger().debug("No masks detected in the image.")
+            return
+
+        instance_seg_img = self.convert_to_instance_seg_img(ret)
+        # Convert to int32 to match 32SC1 encoding expected by cv_bridge
+        instance_seg_img = instance_seg_img.astype(np.int32)
+        msg = Conversions.to_image_msg(header, instance_seg_img, encoding="32SC1")
+        self._pub.publish(msg)
+        self.get_logger().debug("Published instance segmentation image.")
+
+        if self._visualizer is not None:
+            # color_img = self._visualizer.call(ret)
+            color_img = self.visualizer_call(ret, img)
+            self._overlay_pub.publish(
+                Conversions.to_image_msg(header, color_img, encoding="rgb8")
+            )
+
+    def stop(self):
+        """Stop the underlying image worker."""
+        self._worker.stop()
+
+    def convert_to_instance_seg_img(self, ret):
+        """
+        Convert segmentation results to instance segmentation image.
+        Each pixel value encodes both category id and instance id.
+        First 16 bits are category id, last 16 bits are instance id.
+        """
+        masks = ret.masks.cpu().numpy()
+        category_ids = ret.categories.cpu().numpy()
+        img = np.zeros(masks[0].shape, dtype=np.uint32)
+        for i in range(masks.shape[0]):
+            category_id = int(category_ids[i])  # category id are 0-indexed
+            instance_id = i + 1  # instance ids are 1-indexed
+            combined_id = (
+                category_id << 16
+            ) | instance_id  # combine into single uint32
+            img[masks[i, ...] > 0] = combined_id
+
+        return img
```

Comment:
Couple things:
- I'd move this to the non-ros side (and maybe write a quick unit test just to double check the implementation)
- Where the comments are are important, my preference is usually a block comment before the actual code so that the formatting is clearer
- Category IDs being 0-indexed make it a little tricky on the Hydra / Khronos side
- Technically this can be a static method